### PR TITLE
Card #14 View Profile

### PR DESCRIPTION
**Pull request by sayalis-cuelogic**
@onkartibe @aloka-cuelogic 

Card link: https://trello.com/c/u2f9jWMP

Completed view profile feature.

### CODE REVIEW

File: blogauth/forms.py [hunk 1]
```diff
@@ -46,3 +46,9 @@ class LoginForm(auth_forms.AuthenticationForm):
     class Meta:
         model = User
         fields = ['username', 'password']
+
+class PostCountForm(forms.ModelForm):
+   class Meta:
+        model = UserInfo
+        fields = []
+
```
Comment (aloka-cuelogic) [inline blogauth/forms.py]: add new line

File: blogauth/models.py [hunk 2]
```diff
@@ -2,7 +2,9 @@
 from django_countries.fields import CountryField
 from django.core.validators import RegexValidator
 from django.contrib.auth.models import User
-
+from django.db.models.signals import post_save
+from django.db.models import F
```
Comment (aloka-cuelogic) [inline blogauth/models.py]: Where this is used ?

File: blogauth/models.py [hunk 3]
```diff
@@ -24,7 +26,17 @@ class UserInfo(models.Model):
...
+    post_count = models.IntegerField(default=0)
 
     user = models.OneToOneField(User, primary_key=True)
     REQUIRED_FIELDS = ['phone', 'address', 'city', 'gender',
                        'country', 'state', 'dob']
+
+    def __unicode__(self):
+        return self.city
+
+    @receiver(post_save, sender=User)
```
Comment (aloka-cuelogic) [inline blogauth/models.py]: Is this tested ?
Please recheck the logic ?
This should be on save of Post models not on Users.

Please retest and update.

File: blogauth/views.py [hunk 4]
```diff
@@ -20,6 +20,7 @@ def get_queryset(self):
     def post(self, request):
         data = request.POST
         form = RegForm(request.POST)
+        print form.errors
```
Comment (aloka-cuelogic) [inline blogauth/views.py]: Remove all print statement

File: viewprofile/templates/viewprofile/profile.html [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+		            </div>
+		            <div class="col-md-5" id="postcount">
+		                {{userdata.post_count}}
+		            </div>
+		        </div>
+		    </div>
+		</section>
+	</div>
+</div>
+{% endblock%}
```
Comment (aloka-cuelogic) [inline viewprofile/templates/viewprofile/profile.html]: please install pep8 package in your editor

**Comment by aloka-cuelogic** [comment]
@sayalis-cuelogic 

Please implement review comments

**Comment by sayalis-cuelogic** [comment]
Worked on review comments.
Added post_save in post app's models.py

File: post/views.py [hunk 6]
```diff
@@ -86,10 +87,13 @@ def get_queryset(self):
 
     def post(self, request):
         form = PostForm(request.POST)
-        if form.is_valid():
+        user = request.user
+        postcount = PostCountForm(request.POST, instance=user)
```
Comment (aloka-cuelogic) [inline post/views.py]: Don't pass instance from here

File: post/views.py [hunk 7]
```diff
@@ -86,10 +87,13 @@ def get_queryset(self):
...
+        if all([form.is_valid(), postcount.is_valid()]):
             post = form.save(commit=False)
             post.user_id = request.user
             post.save()
+            postcount.save()
```
Comment (aloka-cuelogic) [inline post/views.py]: You don't need to explicitly call this method.
It automatically gets called.

please read more about signals:
http://stackoverflow.com/questions/13014411/django-post-save-signal-implementation
https://docs.djangoproject.com/en/1.8/ref/signals/

File: blogauth/forms.py [hunk 8]
```diff
@@ -46,3 +46,4 @@ class LoginForm(auth_forms.AuthenticationForm):
     class Meta:
         model = User
         fields = ['username', 'password']
+
```
Comment (aloka-cuelogic) [inline blogauth/forms.py]: add new line

File: newsletters/urls.py [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# from django.conf.urls import url
```
Comment (aloka-cuelogic) [inline newsletters/urls.py]: remove commented code.

File: post/models.py [hunk 10]
```diff
@@ -1,16 +1,25 @@
...
-                                     blank=False)
+                                 blank=False)
     post_description = models.CharField(max_length=50000)
     post_category = models.CharField(max_length=100)
     post_is_published = models.BooleanField(default=False)
 
     def __unicode__(self):
         return self.post_name
+
+    @receiver(post_save, sender=User)
```
Comment (aloka-cuelogic) [inline post/models.py]: This is not how post save works.

sender should be what model calls the post save method.
Please do check if created condition as it only should update on create not on edit

**Comment by sayalis-cuelogic** [comment]
Changes done.
Modified models.py of post app.

**Comment by aloka-cuelogic** [comment]
Reviewed Changes.
Merging PR

**Pull request merged**